xapian-core/tests/unittest.cc

```diff
@@ -536,10 +536,22 @@ static bool test_tostring1()
     TEST_EQUAL(str(-1), "-1");
     TEST_EQUAL(str(-9), "-9");
     TEST_EQUAL(str(-10), "-10");
+    TEST_EQUAL(str(0x7f), "127");
+    TEST_EQUAL(str(-0x80), "-128");
+    TEST_EQUAL(str(0x7fff), "32767");
+    // test for lvalue with minimum value for short type.
+    short temp = -32768;
+    TEST_EQUAL(str(temp), "-32768");
+
     TEST_EQUAL(str(0xffffffff), "4294967295");
     TEST_EQUAL(str(0x7fffffff), "2147483647");
     TEST_EQUAL(str(0x7fffffffu), "2147483647");
-    TEST_EQUAL(str(-0x7fffffff), "-2147483647");
```

Comment:
This removes an existing testcase - I think it would be better to just add new testcases since each `str()` call is very quick.